TestMissedRevenueHeatMapGenerator.py

```diff
@@ -0,0 +1,187 @@
+#   (4) business hours for a branch is 9am - 5pm (8 hours)
+
+class TestMissedRevenueHeatmapGenerator(unittest.TestCase):
+#============================================================================================================================================
+    sku = 'FON6-TPD-WHT'
+    branch = 'Cape Town'
+    #use cape town branch business hours
+    #what about public holidays?
+    #closed on sunday
+    normal_business_hours = ['9am', '5pm']
+    weekend_business_hours = ['9am', '1pm']
+    
+    business_hours = {'monday': normal_business_hours,
+                      'tuesday': normal_business_hours,
+                      'wednesday': normal_business_hours,
+                      'thursday': normal_business_hours,
+                      'saturday': weekend_business_hours}
+    
+    time_branch_opens = datetime.datetime(2017,1,3,9)
+    time_branch_closes = datetime.datetime(2017,1,3,17)
+
+    one_week_later = time_branch_opens + datetime.timedelta(days=7)
+    
+    sale_date_1 = time_branch_opens + datetime.timedelta(hours=1)
+    sale_date_2 = time_branch_opens + datetime.timedelta(hours=5)
+    purchase_date_1 = time_branch_opens + datetime.timedelta(hours=2)
+    purchase_date_2 = time_branch_opens + datetime.timedelta(hours=7)
+#============================================================================================================================================
+    def test_calculate_hours_a_branch_is_open(self):
+        '''Test the calculated amount of hours a branch is open'''
+        day_of_the_week = 'monday'
+        #TODO: test for sundays/public holidays
+        #extract information from business_hours dict
+        expected_hours_open = 8
+        calculated_hours_open = MR.calculate_hours_a_branch_is_open(self.branch, day_of_the_week, self.business_hours)
+        
+        self.assertEqual("calculate_hours_a_branch_is_open() returned incorrect value", calculated_hours_open, expected_hours_open)
+    
+    def test_calculate_stock_level_for_sku_at_specific_branch(self):
+        '''Test the calculated amount of stock for a sku at a branch'''
+        #is increment the correct word for stock influencer
+        #need sales and purchases to change stock level
+        #assume stock level is:
+        initial_stock_level = 500
+
+        journal = [{'Branch':self.branch, 'Date':self.sale_date_1, 'Increment':-1,'Sku':self.sku,'Tag':'sale'},
+                   {'Branch':self.branch, 'Date':self.purchase_date_1, 'Increment':3,'Sku':self.sku,'Tag':'purchase'},
+                   {'Branch':self.branch, 'Date':self.sale_date_2, 'Increment':-2,'Sku':self.sku,'Tag':'sale'},
+                   {'Branch':self.branch, 'Date':self.purchase_date_2, 'Increment':5,'Sku':self.sku,'Tag':'purchase'}]
+
+        #500 - 1 + 3 - 2 + 5
+        expected_stock_level = 505
+        calculated_stock_level = MR.test_calculate_stock_level_for_sku_at_specific_branch(self.sku, self.branch, journal, initial_stock_level)
```

Comment:
changed to test_calculate_stock_level_for_sku_at_specific_branch_for_a_period().
Receives time_period as an extra parameter, which effectively calculates the stock level for a certain SKU at a branch over a certain time period